developer-docs/latest/concepts/sso.md

```diff
@@ -0,0 +1,320 @@
+
+A passport strategy is usually built by instantiating it using 2 parameters: the configuration object, and the verify function.
+
+##### Configuration Object
+
+The configuration object depends on the strategy needs, but often asks for a callback URL to be redirected to once the connection has been made on the provider side.
+
+You can generate a specific callback URL for your provider using the `getStrategyCallbackURL` method.
+
+::: tip
+`strapi.admin.services.passport.getStrategyCallbackURL` is a Strapi helper you can use to get a callback URL for a specific provider. It takes a provider name as a parameter and returns a URL as a string.
+:::
+
+If needed, this is also where you will put your client ID and secret key for your OAuth2 application.
+
+##### Verify Function
+
+The verify function is used here as a middleware allowing the user to transform and make extra processing on the data returned from the provider API.
+
+This function always takes a `done` method as last parameter which is used to transfer needed data to the Strapi layer of SSO.
+
+Its signature is the following: `void done(error: any, data: object);` and it follows the following rules:
+
+- If `error` is not set to `null`, then the data sent is ignored, and the controller will throw an error.
+- If the SSO's auto-registration feature is disabled, then the `data` object only need to be composed of an `email` property.
+- If the SSO's auto-registration feature is enabled, then you will need to define (in addition to the `email`) either a `username` property or both `firstname` and `lastname` within the `data` oject.
+
+### Adding a provider
+
+Adding a new provider means adding a new way for your administrators to log-in.
+
+To achieve a great flexibility and a large choice of provider, Strapi uses [Passport.js](http://www.passportjs.org/). Any valid passport strategy that doesn't need additional custom data should therefore work with Strapi.
+
+::: warning
+Strategies such as [ldapauth](https://github.com/vesse/passport-ldapauth) don't work out of the box since they require extra data to be sent from the admin panel.
+If you want to add an LDAP provider to your application, you will need to write a [custom strategy](http://www.passportjs.org/packages/passport-custom/).
+You can also use services such as Okta and Auth0 as bridge services.
+:::
+
+### Configuring the provider
+
+To configure a provider, follow the procedure below:
+
+1. Make sure to import your strategy in your server configuration file, either from an installed package or a local file.
+2. You'll need to add a new item to the `admin.auth.providers` array in your server configuration that will match the [format given above](#provider-configuration)
+3. Restart your application, the provider should appear on your admin login page.
+
+## Examples
+
+:::: tabs
+
+::: tab Google
+
+Using: [passport-google-oauth2](https://github.com/mstade/passport-google-oauth2)
+
+```bash
+yarn add passport-google-oauth2
```

Comment:
We should probably clarify yarn vs npm at some point to harmonize our documentation. Sometimes we use npm, sometimes yarn, sometimes both. :thinking:

Reply:
You're right, I'll put both methods here